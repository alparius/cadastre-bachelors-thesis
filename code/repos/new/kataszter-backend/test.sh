#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
