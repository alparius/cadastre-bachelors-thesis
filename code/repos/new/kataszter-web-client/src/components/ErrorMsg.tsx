import React from "react";
import { FormattedMessage as FM } from "react-intl";
import { NavLink } from "react-router-dom";
import { Container, Header } from "semantic-ui-react";

interface IErrorMsgProps {
    errorMessage?: string;
}

const ErrorMsg: React.FC<IErrorMsgProps> = (props: IErrorMsgProps) => {
    const { errorMessage } = props;

    return (
        <Container style={{ marginTop: "7em" }}>
            <Header as="h2" className="red">
                {errorMessage ? <FM id={errorMessage} /> : <FM id="error.basic" />}
            </Header>

            <Header as="h3">
                <NavLink to="/" className="hoverLink">
                    <FM id="error.backHome" />
                </NavLink>
            </Header>
        </Container>
    );
};

export default ErrorMsg;
