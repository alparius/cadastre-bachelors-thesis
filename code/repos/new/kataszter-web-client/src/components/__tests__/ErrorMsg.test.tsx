/*===================
  ErrorMsg.unit.test.tsx
 ===================*/
import React from "react";

import { createComponentWithIntlAndRouter } from "../../util/test/test-wrapper";
import ErrorMsg from "../ErrorMsg";

describe("ErrorMsg element", () => {
    it("should render a container with a header containing the error message", () => {
        const container = createComponentWithIntlAndRouter(<ErrorMsg errorMessage="test" />).toJSON();
        expect(container).toMatchSnapshot();
    });
});
