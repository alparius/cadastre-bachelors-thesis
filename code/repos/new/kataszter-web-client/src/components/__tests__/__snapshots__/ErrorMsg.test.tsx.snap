// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMsg element should render a container with a header containing the error message 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "7em",
    }
  }
>
  <h2
    className="ui header red"
  >
    test
  </h2>
  <h3
    className="ui header"
  >
    <a
      aria-current="page"
      className="hoverLink active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Back to the Home Page
    </a>
  </h3>
</div>
`;
